### PR TITLE
Lbatch/add update package versions

### PR DESCRIPTION
**Pull request by lukebatchelor**
Adds the updatePackageVersions util function for bumping/releasing

### CODE REVIEW

**Comment by thejameskyle** [comment]
You should export this from https://github.com/pyarn/pyarn/blob/master/src/index.js

File: src/utils/updatePackageVersions.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      const depType = pkg.getDependencyType(depName);
+      if (!depType) continue;
+
+      promises.push(pkg.updateDependencyVersionRange(depName, depType, versions[depName]));
+    }
+
+    return Promise.all(promises);
+  }
+
+  await runWorkspaceTasks(updateWorkspacePackageVersion, opts);
```
Comment (thejameskyle) [inline src/utils/updatePackageVersions.js]: You don't need `runWorkspaceTasks` for this, it's meant only for scripting.

Comment (lukebatchelor) [reply src/utils/updatePackageVersions.js]: Should this take in a Workspace then?

Comment (thejameskyle) [reply src/utils/updatePackageVersions.js]: Nah, we should not expose internal objects like `Workspace` or `Package`. This is all you need to do:

```
let cwd = process.cwd();
let project = await Project.init(cwd);
let workspaces = await project.getWorkspaces();

for (let workspace of workspaces) {
  // ...
}
```

File: src/utils/updatePackageVersions.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  for (let workspace of workspaces) {
+    const pkg = workspace.pkg;
+    const promises = [];
+
+    for (let depName of Object.keys(versions)) {
+      const depType = pkg.getDependencyType(depName);
+      if (!depType) continue;
+
+      promises.push(pkg.updateDependencyVersionRange(depName, depType, versions[depName]));
```
Comment (thejameskyle) [inline src/utils/updatePackageVersions.js]: This should probably be:

```js
'^' + versions[depName]
```

That way the version ranges that we're publishing can match a wider range

**Comment by thejameskyle** [comment]
Fix that one thing and feel free to merge

File: src/utils/updatePackageVersions.js (hunk 2 above)
Comment (lukebatchelor) [reply src/utils/updatePackageVersions.js]: Oh, in testing, I was adding the hat in myself.

**Pull request merged**